sdk_extensions/jfr_events/src/main/java/io/opentelemetry/sdk/extension/jfr/JfrContextStorageProvider.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.jfr;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextStorage;
+import io.opentelemetry.context.ContextStorageProvider;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.trace.Span;
+
+public class JfrContextStorageProvider implements ContextStorageProvider {
+
+  @Override
+  public ContextStorage get() {
+    ContextStorage parentStorage = ContextStorage.get();
```

Comment:
@jkwatson is this how I should do things to wrap the an existing ContextStorage?

What would be the best way to have this provider detected and enabled as a wrapper when JAR is added on the classpath? Or should it be done through some configuration/code instead?

Reply:
hmm. I don't know if this is right. I don't think you want to return a new storage for each call, but the same storage. Take a look at the `GrpcContextStorageProvider` and `BraveContextStorageProvider` for examples. @anuraaga is definitely the expert here, if you have questions after looking at them.

Reply:
thanks, will check them. I just quickly looked at the example that was in ContextStorage which was basically done this way